boards/beagle/beagleplay_cc1352/beagleplay_cc1352_defconfig

```diff
@@ -0,0 +1,23 @@
+#
+# Copyright (c) 2022 Jason Kridner, BeagleBoard.org Foundation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+CONFIG_GPIO=y
+CONFIG_PINCTRL=y
+CONFIG_PINCTRL_DYNAMIC=y
+CONFIG_PINCTRL_CC13XX_CC26XX=y
```

Comment:
this is wrong, where is dynamic used? why is a driver impl enabled? why is pinctrl even enabled here?

Reply:
I have removed dynamic and drive impl option. Pinctrl is required for board antenna init.

Reply:
pinctrl needs to be selected by drivers, not enabled manually like this.

Reply:
So, the board_antenna is not a driver but platform init stuff. So where should I enable PINCTRL? Currently, `beagleconnect_freedom`, `cc1352p1_launchxl` also enable pinctrl in defconfig.

Reply:
I have only left `CONFIG_PINCTRL=y` in the defconfig like most other current boards seem to do and removed `CONFIG_PINCTRL_DYNAMIC=y` and `CONFIG_PINCTRL_CC13XX_CC26XX=y`. Feel free to tell if `CONFIG-PINCTRL` is not supposed to go here in new boards.